registry/storage/linkedblobstore.go

```diff
@@ -95,10 +97,56 @@ func (lbs *linkedBlobStore) Put(ctx context.Context, mediaType string, p []byte)
 	return desc, lbs.linkBlob(ctx, desc)
 }
 
+type createOptions struct {
```

Comment:
Currently the BlobCreateOptions are intended to be extensible based on the implementation of the BlobIngester, and the implementation may choose to honor any or none of the BlobCreateOptions passed in. For the blob mounting operation we currently have a LinkedBlobStore implementation and the registry client both needing similar options (a source repository and a digest), so these could either use a shared type (not sure where to define it) or they can continue to use similar (or identical) options.